gravityflow/gravityflow
# Added filters to support GPPA hydration

## Description
See [HS#17596](https://secure.helpscout.net/conversation/1545379159/17596/) / [HS#18084](https://secure.helpscout.net/conversation/1597426946/18084?folderId=1776095). After integration testing with PR #420 / GPPA - the similar hydration issue was identified in ActiveCampaign step that would have potential for all step types. This PR provides a steptype-specific-named filter - gravityflow_{$this->get_type()}_form - and a generic named filter - gravityflow_step_form - to allow hydration to be applied at one or more levels. Or other scenarios where the form needs customization before/during step processing. This PR also addresses the same issues as #420 through the step-type based filter (gravityflow_notification_form)

## Testing instructions
1. Add a form with a single text field an a radio field.
1. Using GPPA, set the radio field to use dynamic choices.
1. Preview the form, and submit it with the default Notification enabled.
1. Check the email and ensure you see the Label for the value you chose in the Radio field.
1. Now edit the form settings and disable the default notification.
1. Add a Workflow Notification step, and choose the default admin notification as the one to send.
1. Submit the form again.
1. Check the email and ensure you still see the Label being displayed, and not the value.
1. Receive an updated version of GPPA that uses uses the gravityflow_step_form filter to hydrate.
1. Perform similar steps as listed above with multiple feed add-ons or other step types.

## Automated Test Enhancements
None
 
## Documentation Changes
New doc pages for two filters will be required.

## Checklist:
